### DISCUSSION TITLE
Prevent uncommitted objects from S3 access

### DISCUSSION THREAD

**Discussion by dsgibbons** [treeverse/lakeFS#6286] [Q&A] [answered]
In my current lakeFS installation, I am able to download objects using an S3 client that have not been committed yet. That is, I follow the workflow:

1. Upload object "object-name" to repo via `lakectl fs upload ...`
2. Download "object-name" via MinIO Client `mc cp ...`

I would have expected to receive an _Object does not exist_ error until "object-name" has been committed. Are there any setup options to enforce this behavior?

**Comment by arielshaqed** [comment, marked as answer]
Hi @dsgibbons ,
Let me try to help...
lakeFS has s3 semantics: if you upload an object, it is immediately available to download. This is required for almost any s3-based program to function.
But obviously as you say you will sometimes want to see only committed data. You can use more advanced commit refs to do that! For instance, `main@` refers to committed objects at main, which IIUC should be exactly what you need right now. You can read more about reference expressions [here](https://docs.lakefs.io/understand/model.html#ref-expressions).
HTH!

**Reply by dsgibbons** [reply]
`main@` works perfectly! Thank you for the fast response. I note however that the provided link does not mention the `@` operator. Is there any specific documentation that references `@`?
